### PR TITLE
Add amazon.com test

### PR DESCRIPTION
**Pull request by emmaaghlamazyan94**

### CODE REVIEW

File: src/main/java/pages/AuthorPage.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    public AuthorPage(WebDriver driver) {
+        this.driver = driver;
+        this.wait = new WebDriverWait(driver, 10);
+    }
+
+    public void clickOnSearchResult() {
+        List<WebElement> authors = driver.findElements
+                (By.xpath("//div[@class='a-row a-size-base a-color-secondary']//a[@class='a-size-base a-link-normal']"));
+        for (WebElement authorLink : authors) {
```
Comment (chandranik90) [inline src/main/java/pages/AuthorPage.java:23]: What's the reason of finding list of items, click on the first and break the loop?

File: src/test/java/AmazonTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    public void findAndClickOnAuthorBooks() {
+        HomePage searchPage = new HomePage(driver);
+        searchPage.waitUntilPageLoads();
+        searchPage.searchAuthorName(authorFullName, "Books");
+        BooksPage booksPage = new BooksPage(driver);
+        booksPage.waitUntilPageLoads();
+        AuthorPage authorPage = new AuthorPage(driver);
+        authorPage.clickOnSearchResult();
```
Comment (chandranik90) [inline src/test/java/AmazonTest.java:66]: Seems, this is not an `AuthorPage` yet. Logically, you are still on `Books Page` when clicking on found result. Only after that you navigate to `Author Page`. So, the `clickOnSearchResult` method should be moved to `Books Page`

File: src/main/java/pages/AuthorPage.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            wait.until(ExpectedConditions.elementToBeClickable(authorLink));
+            authorLink.click();
+            break;
+        }
+    }
+
+    public void clickToSort() {
+        WebElement sortBy = driver.findElement(By.xpath("//span//span[@class='a-button-text a-declarative']"));
+        sortBy.click();
+        WebElement lowToHigh = driver.findElement(By.id("dropdown1_1"));
```
Comment (chandranik90) [inline src/main/java/pages/AuthorPage.java:33]: Bad locator. Use text to find the option in dropdown

File: src/main/java/pages/AuthorPage.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        lowToHigh.click();
```
Comment (chandranik90) [inline src/main/java/pages/AuthorPage.java:34]: This action brings to page reload. So, it's necessary to wait for page to load

File: src/main/java/pages/AuthorPage.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public boolean sortByPrice() {
+        List<WebElement> prices = driver.findElements(By.xpath("//span[@class='a-price']//span[@aria-hidden='true']"));
+        ArrayList<Float> priceList = new ArrayList();
```
Comment (chandranik90) [inline src/main/java/pages/AuthorPage.java:39]: Use `<>` operator when initializing generic class

File: src/main/java/pages/AuthorPage.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        for (int i = 0; i < prices.size(); i = i++) {
```
Comment (chandranik90) [inline src/main/java/pages/AuthorPage.java:40]: is the `i = i++` really what you want to achieve? if yes, could you please explain?

File: src/main/java/pages/AuthorPage.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    public void clickToSort() {
+        WebElement sortBy = driver.findElement(By.xpath("//span//span[@class='a-button-text a-declarative']"));
+        sortBy.click();
+        WebElement lowToHigh = driver.findElement(By.id("dropdown1_1"));
+        lowToHigh.click();
+    }
+
+    public boolean sortByPrice() {
+        List<WebElement> prices = driver.findElements(By.xpath("//span[@class='a-price']//span[@aria-hidden='true']"));
```
Comment (chandranik90) [inline src/main/java/pages/AuthorPage.java:38]: This locator doesn't find anything for me. Please recheck it

File: src/main/java/pages/AuthorPage.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        ArrayList<Float> priceList = new ArrayList();
+        for (int i = 0; i < prices.size(); i = i++) {
+            priceList.add(Float.parseFloat(prices.get(i).getText()));
+        }
+        if (isListSorted(priceList)) {
```
Comment (chandranik90) [inline src/main/java/pages/AuthorPage.java:43]: Simply replace by `return isListSorted(priceList)`

**Review by chandranik90 — CHANGES_REQUESTED** [review]

File: src/main/java/pages/AuthorPage.java (hunk 6 above)
Comment (emmaaghlamazyan94) [reply src/main/java/pages/AuthorPage.java:40]: No, it was just misprint

File: src/main/java/pages/AuthorPage.java (hunk 1 above)
Comment (emmaaghlamazyan94) [reply src/main/java/pages/AuthorPage.java:24]: Not all results are hyperlinks, some are only the name of the author, so I find the first clickable element, click on it and break.

File: src/main/java/pages/BooksPage.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+                checkCondition = true;
+            } else {
+                return false;
+            }
+        }
+        return checkCondition;
+    }
+
+    public void waitUntilPageLoads() {
+        wait.until(ExpectedConditions.numberOfElementsToBeMoreThan
```
[the remark this answers is not in the document]
Comment (emmaaghlamazyan94) [reply src/main/java/pages/BooksPage.java:34]: Is it good to block wait with try catch? And if there is no books found, catch the exception?

File: src/test/java/AmazonTest.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        BooksPage booksPage = new BooksPage(driver);
+        booksPage.waitUntilPageLoads();
+        Assert.assertTrue(checkAllBooksBySameAuthor(authorFullName),
+                "Not all books are written by the searched author " + authorFullName);
+    }
+
+    @Test
+    public void confirmSearchedTextExistence() {
+        findAndClickOnAuthorBooks();
+        String textToSearch = ("Books by " + authorFullName).toLowerCase();
```
Comment (chandranik90) [inline src/test/java/AmazonTest.java:50]: Still the same. This is your text. And you could just write it in lowercase initially

**Review by chandranik90 — APPROVED** [review]

**Pull request merged**